infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultAdapterForUpgradeActive.java

```diff
@@ -0,0 +1,110 @@
+package org.corfudb.infrastructure.logreplication.infrastructure.plugins;
+
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.view.TableRegistry;
+import org.corfudb.utils.CommonTypes;
+import org.corfudb.utils.LogReplicationStreams;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * This class is created for LR rolling upgrade tests to serve as the plugin for Active cluster to
+ * get the set of streams to replicate.
+ */
+public class DefaultAdapterForUpgradeActive implements ILogReplicationConfigAdapter {
```

Comment:
The code between DefaultAdapterForUpgradeActive and DefaultAdapterForUpgradeStandby is basically the same, seems only thing changing is the endpoint, so it makes sense to avoid code duplication and create a base class with an abstract method getEndpoint()